homeassistant/components/loqed/config_flow.py

```diff
@@ -41,27 +95,62 @@ async def async_step_zeroconf(
         # Check if already exists
         await self.async_set_unique_id(lock_data["bridge_mac_wifi"])
         self._abort_if_unique_id_configured({CONF_HOST: host})
-        return await self.async_step_user()
 
-    async def async_oauth_create_entry(self, data: dict) -> FlowResult:
-        """Create an entry for the flow.
+        return await self.async_step_user()
 
-        Ok to override if you want to fetch extra info or even add another step.
-        """
-        session = async_get_clientsession(self.hass)
-        res = await session.request(
-            "GET",
-            "https://app.loqed.com/API/integration_oauth3/retrieve_lock_data.php",
-            headers={"Authorization": f"Bearer {data['token']['access_token']}"},
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Show userform to user."""
+        user_data_schema = vol.Schema(
+            {
+                vol.Required(CONF_API_TOKEN): str,
+            }
         )
-        lock_data = await res.json(content_type="text/html")
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=user_data_schema,
+                description_placeholders={
+                    "config_url": "https://integrations.production.loqed.com/personal-access-tokens",
+                },
+            )
 
-        await self.async_set_unique_id(
-            re.sub(
-                r"LOQED-([a-f0-9]+)\.local", r"\1", lock_data["bridge_mdns_hostname"]
+        errors = {}
+
+        try:
+            info = await self.validate_input(self.hass, user_input)
+        except CannotConnect:
+            errors["base"] = "cannot_connect"
+        except InvalidAuth:
+            errors["base"] = "invalid_auth"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+        else:
+            await self.async_set_unique_id(info["id"])
+            self._abort_if_unique_id_configured()
+
+            return self.async_create_entry(
+                title="LOQED Touch Smart Lock",
+                data=(
+                    user_input | {CONF_WEBHOOK_ID: webhook.async_generate_id()} | info
+                ),
             )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=user_data_schema,
+            errors=errors,
+            description_placeholders={
+                "config_url": "https://integrations.production.loqed.com/personal-access-tokens",
```

Comment:
See my previous comment. And if it has a valid reason to be user configurable then I'd suggest to move the default value into a constant.